### PR TITLE
Devops

### PR DESCRIPTION
**Pull request by benjamistan**
Added buildspec.yml file to facilitate AWS CI/CD Pipeline.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+```diff
+- This is a DevOps ephemeral branch for configuring the CI/CD in AWS. There are no new features or code.
```
Comment (v-stickykeys) [inline README.md:2]: Not sure I understand this... This pr is being merged into develop so this message isn't applicable. Anyway dont think we need different messages on different branches. And dont need diff branch for ci/cd.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+```diff
+- This is a DevOps ephemeral branch for configuring the CI/CD in AWS. There are no new features or code. 
+- Nothing to see here. To use GodMode-Ganache-CLI, the Develop branch is where you want to be
```
Comment (v-stickykeys) [inline README.md:3]: If needed let's just make this more clear like, `Use this library from `develop` branch`

File: buildspec.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+version: 0.2
+
+env:
+  parameter-store:
+    AWS_ACCOUNT_ID: account_id
```
Comment (v-stickykeys) [inline buildspec.yml:5]: How does this work? Is it coming from github secrets? If not that's an easy way to get this.

File: buildspec.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  post_build:
+    commands:
+      - echo Build completed on `date`
+      - echo Pushing the Docker image...
+      - docker push $REPOSITORY_URI:latest
+      - docker push $REPOSITORY_URI:$IMAGE_TAG
+      - echo Writing image definitions file...
+      - printf '[{"name":"godmode-ganache-cli","imageUri":"%s"}]' $REPOSITORY_URI:$IMAGE_TAG > imagedefinitions.json
+artifacts:
+  files: imagedefinitions.json
```
Comment (v-stickykeys) [inline buildspec.yml:35]: Do we need an artifact? What is it used for?

**Review by v-stickykeys — CHANGES_REQUESTED** [review]
Great overall!

Besides the code questions a few clean up things:

- Make your commit messages more descriptive so that we actually know how each commit changes behavior of the pipeline

- If the commits feel like they should have the same message do an interactive rebase and just fold them into a single commit

- Since we do have the account id in commit history right now do an interactive rebase so that the account id gets removed from git history

This is a reference that I like for rebasing https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history#squash-commits-together

File: buildspec.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      - COMMIT_HASH=$(echo $CODEBUILD_RESOLVED_SOURCE_VERSION | cut -c 1-7)
+      - echo Commit Hash is $COMMIT_HASH
+      - IMAGE_TAG=${COMMIT_HASH:=latest}
+      - echo Image Tag is $IMAGE_TAG
+  build:
+    commands:
+      - echo Build started on `date`
+      - echo Building the Docker image...
+      - docker build -t $REPOSITORY_URI:latest .
+      - docker tag $REPOSITORY_URI:latest $REPOSITORY_URI:$IMAGE_TAG
```
Comment (v-stickykeys) [inline buildspec.yml:25]: Can we also include branch name as a tag?

**Comment by benjamistan** [comment]
Closing. This is a mess indeed. At this point it's more productive for me to try again from scratch than to unpick this chaos.

**Pull request closed without merging**